metadata-ingestion/src/datahub/ingestion/source/glue.py

```diff
@@ -131,44 +151,270 @@ def __init__(self, config: GlueSourceConfig, ctx: PipelineContext):
         self.source_config = config
         self.report = GlueSourceReport()
         self.glue_client = config.glue_client
+        self.s3_client = config.s3_client
+        self.extract_transforms = config.extract_transforms
         self.env = config.env
 
     @classmethod
     def create(cls, config_dict, ctx):
         config = GlueSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
-    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
-        def get_all_tables() -> List[dict]:
-            def get_tables_from_database(database_name: str) -> List[dict]:
-                new_tables = []
-                paginator = self.glue_client.get_paginator("get_tables")
-                for page in paginator.paginate(DatabaseName=database_name):
-                    new_tables += page["TableList"]
-
-                return new_tables
-
-            def get_database_names() -> List[str]:
-                database_names = []
-                paginator = self.glue_client.get_paginator("get_databases")
-                for page in paginator.paginate():
-                    for db in page["DatabaseList"]:
-                        if self.source_config.database_pattern.allowed(db["Name"]):
-                            database_names.append(db["Name"])
-
-                return database_names
-
-            if self.source_config.database_pattern.is_fully_specified_allow_list():
-                database_names = self.source_config.database_pattern.get_allowed_list()
+    def get_all_jobs(self):
+        """
+        List all jobs in Glue.
+        """
+
+        jobs = []
+
+        paginator = self.glue_client.get_paginator("get_jobs")
+        for page in paginator.paginate():
+            jobs += page["Jobs"]
+
+        return jobs
+
+    def get_dataflow_graph(self, script_path: str) -> Dict[str, Any]:
+        """
+        Get the DAG of transforms and data sources/sinks for a job.
+
+        Parameters
+        ----------
+            script_path:
+                S3 path to the job's Python script.
+        """
+
+        # extract the script's bucket and key
+        url = urlparse(script_path, allow_fragments=False)
+        bucket = url.netloc
+        key = url.path[1:]
+
+        # download the script contents
+        obj = self.s3_client.get_object(Bucket=bucket, Key=key)
+        script = obj["Body"].read().decode("utf-8")
+
+        # extract the job DAG from the script
+        return self.glue_client.get_dataflow_graph(PythonScript=script)
```

Comment:
interesting. didn't realize you are getting the graph from the script itself..

Reply:
Yeah, I thought they would have a method for getting it directly, but seems like this is the only way to get it through boto